spec/requests/application_controller_spec.rb

```diff
@@ -33,4 +33,67 @@
     end
   end
 
+  describe "#handle_theme" do
+    let(:theme) { Theme.create!(user_id: -1, name: 'bob', user_selectable: true) }
+    let(:theme2) { Theme.create!(user_id: -1, name: 'bobbob', user_selectable: true) }
+    let(:user) { Fabricate(:user) }
+    let(:admin) { Fabricate(:admin) }
+
+    before do
+      sign_in(user)
+    end
+
+    it "selects the theme the user has selected" do
+      user.user_option.update_columns(theme_ids: [theme.id])
+
+      get "/"
+      expect(controller.theme_ids).to eq([theme.id])
+
+      theme.update_attribute(:user_selectable, false)
+
+      get "/"
+      expect(controller.theme_ids).not_to eq([theme.id])
```

Comment:
We almost always want to avoid using `not_to eq()` because the result value can be anything and the test will still pass. Instead, we want the assertion to be tested against the actual value that we expect.